src/test/java/io/wgj/ctci/ArrayListTest.java

```diff
@@ -4,132 +4,108 @@
+    for (Integer i : ints) {
+      list.add(i);
     }
-    assertEquals(4, al.size());
+    assertEquals(ints[0], list.remove(0));
+    Integer[] ints_with_front_removed = {2, 3, 4, 5};
+    assertTrue(compareArrayToArrayList(ints_with_front_removed, list));
   }
 
   @Test
   public void arrayList_remove_middle() {
-    ArrayList<Integer> al = new ArrayList<>();
-    Integer[] add_ints = {1, 2, 3, 4, 5};
-    for (Integer i : add_ints) {
-      al.add(i);
-    }
-    assertEquals(add_ints[2], al.remove(2));
-
-    Integer[] get_ints = {1, 2, 4, 5};
-    for (int i = 0; i <= get_ints.length - 1; i++) {
-      assertEquals(get_ints[i], al.get(i));
+    ArrayList<Integer> list = new ArrayList<>();
+    Integer[] ints = {1, 2, 3, 4, 5};
+    for (Integer i : ints) {
+      list.add(i);
     }
-    assertEquals(4, al.size());
+    assertEquals(ints[2], list.remove(2));
+    Integer[] ints_with_middle_removed = {1, 2, 4, 5};
+    assertTrue(compareArrayToArrayList(ints_with_middle_removed, list));
   }
 
   @Test
   public void arrayList_remove_back() {
-    ArrayList<Integer> al = new ArrayList<>();
-    Integer[] add_ints = {1, 2, 3, 4, 5};
-    for (Integer i : add_ints) {
-      al.add(i);
-    }
-    assertEquals(add_ints[4], al.remove(4));
-
-    Integer[] get_ints = {1, 2, 3, 4};
-    for (int i = 0; i <= get_ints.length - 1; i++) {
-      assertEquals(get_ints[i], al.get(i));
+    ArrayList<Integer> list = new ArrayList<>();
+    Integer[] ints = {1, 2, 3, 4, 5};
+    for (Integer i : ints) {
+      list.add(i);
     }
-    assertEquals(4, al.size());
+    assertEquals(ints[4], list.remove(4));
+    Integer[] ints_with_back_removed = {1, 2, 3, 4};
+    assertTrue(compareArrayToArrayList(ints_with_back_removed, list));
   }
+
   @Test
   public void arrayList_isEmpty() {
-    ArrayList<Integer> al = new ArrayList<>();
-    assertTrue(al.isEmpty());
-    al.add(0);
-    assertFalse(al.isEmpty());
-    al.remove(0);
-    assertTrue(al.isEmpty());
+    ArrayList<Integer> list = new ArrayList<>();
+    assertTrue(list.isEmpty());
+    list.add(0);
+    assertFalse(list.isEmpty());
+    list.remove(0);
+    assertTrue(list.isEmpty());
   }
+
   @Test
   public void arrayList_set() {
-    ArrayList<Integer> al = new ArrayList<>();
-    al.add(0);
-    assertEquals((Integer) 0, al.get(0));
-    al.set(0, 42);
-    assertEquals((Integer) 42, al.get(0));
+    ArrayList<Integer> list = new ArrayList<>();
+    list.add(0);
+    assertEquals((Integer) 0, list.get(0));
+    list.set(0, 42);
+    assertEquals((Integer) 42, list.get(0));
   }
+
   @Test
   public void arrayList_Iterator() {
-    ArrayList<Integer> al = new ArrayList<>();
+    ArrayList<Integer> list = new ArrayList<>();
     Integer[] test = {1, 2, 3, 4, 5};
     for (Integer i : test) {
-      al.add(i);
+      list.add(i);
     }
     int index = 0;
-    for (Integer i : al) {
-      assertEquals(al.get(index), i);
+    for (Integer i : list) {
+      assertEquals(list.get(index), i);
```

Comment:
This also needs to verify that the length of items the iterator returns matches the expected length, I believe there is a bug here.